WordPress/src/main/java/org/wordpress/android/ui/mysite/SelectedSiteSource.kt

```diff
@@ -9,12 +15,40 @@ import javax.inject.Singleton
 
 @Singleton
 class SelectedSiteSource @Inject constructor(
-    private val selectedSiteRepository: SelectedSiteRepository
+    private val selectedSiteRepository: SelectedSiteRepository,
+    private val dispatcher: Dispatcher
 ) : MySiteSource<SelectedSite> {
+    val refresh: MutableLiveData<Boolean> = MutableLiveData<Boolean>(false)
+
     override fun buildSource(
         coroutineScope: CoroutineScope,
         siteLocalId: Int
     ) = selectedSiteRepository.selectedSiteChange
             .filter { it == null || it.id == siteLocalId }
             .map { SelectedSite(it) }
+
+    fun refresh() {
+        selectedSiteRepository.updateSiteSettingsIfNecessary()
+        selectedSiteRepository.getSelectedSite()?.let {
+            refresh.postValue(true)
+            dispatcher.dispatch(SiteActionBuilder.newFetchSiteAction(it))
+        }
+    }
+
+    fun isRefreshing() = refresh.value as Boolean
+
+    init {
+        dispatcher.register(this)
+    }
+
+    fun clear() {
+        dispatcher.unregister(this)
+    }
+
+    @SuppressWarnings("unused")
```

Comment:
Minor (🔍): You can suppress using the `@Suppress` annotation, and with that you can also suppress the unused parameter for now so that it doesn't resolve to a warning, like that: `@Suppress("unused", "UNUSED_PARAMETER")`